chainer/functions/math/zeta.py

```diff
@@ -0,0 +1,62 @@
+from chainer.backends import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+
+_zeta_cpu = None
+
+
+class Zeta(function_node.FunctionNode):
+
+    @property
+    def label(self):
+        return 'zeta'
+
+    def check_type_forward(self, in_types):
+        type_check._argname(in_types, ('x', 'q'))
+        x_type, q_type = in_types
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            q_type.dtype.kind == 'f',
+        )
+
+    def forward_cpu(self, inputs):
+        x, q = inputs
+        global _zeta_cpu
+        if _zeta_cpu is None:
+            try:
+                from scipy import special
+                _zeta_cpu = special.zeta
+            except ImportError:
+                raise ImportError('Scipy is not available. Forward computation'
+                                  ' of zeta cannot be done.')
+        self.retain_inputs((0, 1))
+        return utils.force_array(_zeta_cpu(x, q), dtype=x.dtype),
+
+    def forward_gpu(self, inputs):
+        x, q = inputs
+        self.retain_inputs((0, 1))
+        return utils.force_array(
+            cuda.cupyx.scipy.special.zeta(x, q), dtype=x.dtype),
+
+    def backward(self, indexes, gy):
+        x, q = self.get_retained_inputs()
+        return zeta(x - 1, q) * -gy[0], zeta(x - 1, q) * -gy[0],
```

Comment:
Hm, is this backward computation correct? https://dlmf.nist.gov/25.11

Reply:
Yeah my bad 😅 it's corrected now. Have a look!

Reply:
Thanks that looks better, but the gradient you're computing is for `q`. It can be `None` for `x`?

Reply:
When I pass `None` as a parameter, it throws `NoneType has no attribute shape`

Reply:
How do you suggest I fix that? I fixed it by arbitrarily passing the same value